eclair-core/src/main/scala/fr/acinq/eclair/offer/OfferManager.scala

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2023 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.offer
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import fr.acinq.bitcoin.scalacompat.Crypto.{PrivateKey, PublicKey}
+import fr.acinq.bitcoin.scalacompat.{ByteVector32, ByteVector64, Crypto}
+import fr.acinq.eclair.db.{IncomingBlindedPayment, IncomingPaymentStatus, PaymentType}
+import fr.acinq.eclair.message.{OnionMessages, Postman}
+import fr.acinq.eclair.payment.receive.MultiPartHandler
+import fr.acinq.eclair.payment.receive.MultiPartHandler.{CreateInvoiceActor, ReceivingRoute}
+import fr.acinq.eclair.payment.{Bolt12Invoice, Invoice}
+import fr.acinq.eclair.wire.protocol.CommonCodecs.{bytes32, bytes64, lengthPrefixedFeaturesCodec, millisatoshi, publicKey, timestampSecond, uint64overflow}
+import fr.acinq.eclair.wire.protocol.OfferTypes.{InvoiceRequest, InvoiceTlv, Offer}
+import fr.acinq.eclair.wire.protocol.PaymentOnion.FinalPayload
+import fr.acinq.eclair.wire.protocol.{GenericTlv, MessageOnion, OfferTypes, OnionMessagePayloadTlv, TlvStream}
+import fr.acinq.eclair.{Feature, Features, MilliSatoshi, NodeParams, TimestampMilli, TimestampSecond, randomBytes32}
+import scodec.bits.ByteVector
+import scodec.codecs._
+import scodec.{Attempt, Codec, DecodeResult}
+
+import scala.concurrent.duration.DurationInt
+import scala.util.{Success, Try}
+
+object OfferManager {
+  // @formatter:off
+  sealed trait Command
+  case class RegisterOffer(offer: Offer, nodeKey: PrivateKey, pathId_opt: Option[ByteVector32], handler: ActorRef[HandlerCommand]) extends Command
```

Comment:
It would be useful to document that the `pathId_opt` here is what has been included in the blinded paths associated with the offer and must be verified when receiving `invoice_request`s, otherwise it's easy to get confused between which `path_id`s are associated to what (this is different from the `path_id` that will be used for the payments, right?).

Reply:
Done